### ISSUE TITLE
Fixing Bucklespring for Windows

### ISSUE THREAD

**Issue by vipulgupta2048** [zevv/bucklespring#68]
Hi @zevv, First of all, I love Bucklesping and I am willing to try and fix it for Windows. Like you I too, don't know much about Windows and its development. All my development goes on Linux too. So if you can help me get started then I can probably get it fixed. 
- What is broken? 
- Any fixes you have in mind?
- What all should I learn?
- A little about the code you already wrote? 
Also, I have taken this on as a project at my college so if this doesn't work out. I might find another way of making the same thing and packaging it in .exe

**Comment by zevv** [comment]
Quoting Vipul Gupta (2018-05-11 08:36:04)

> - What is broken? 

I'm not exactly sure, but if I recall correctly it has something to do
with the OpenAL library used by bucklespring, my notes say something
about `alureCreateBufferFromFile` and threads, that's all I know :)

> - Any fixes you have in mind?

No idea - I do not have machines running Windows in my neighbourhood,
and I never seem to be able to do simple debugging on windows with
simple tools (i.e. gdb), so I'm not even sure where to start.

> - What all should I learn?

To get started you could first build and run buckle on Linux to see
things working as they should - there's not much to the code: just
reading events from the keyboard, and playing a sample with OpenAL.
Then build for windows, and see what works and what does not. I believe
the keyboard handling is just fine (scan-windows.c), and the problem
is somewhere in the interaction with the OpenAL lib.

Good luck!

--
:wq
^X^Cy^K^X^C^C^C^C

**Comment by Kamek437** [comment]
Necro. I wish this worked on windows. All the other alternatives are waaaay to heavy on resources for me. I don't want to run a full electron instance just for keyboard sounds. I know it's nota  high priority but would love to use it on windows. Got used to it on my NixOS box.

**Comment by KreaeTerra** [comment]
Yeah I'd hate to necro but this used to be my favorite mostly because of the directional keypress audio other similar software seems to lack. Wish someone could fix it.

**Comment by jalbarl** [comment]
I recently built bucklespring on Windows using VS Build Tools 2022 (32-bit: openal-soft-1.23.1 official binaries + alure-1.2 official binaries; 64-bit: openal-soft-1.23.1 official binaries + alure-1.x latest branch) and I am not experiencing any crashes, missing sound or odd behavior.

I had some build struggles due to the missing getopt library and I had to create my own Makefile, but otherwise it seems to Just Work™.

**Comment by Kamek437** [comment]
Creating your own makefiles isn't "just works".

**Comment by jalbarl** [comment]
> Creating your own makefiles isn't "just works".

That was only needed in my specific setup, because I wanted to build bucklespring using the Microsoft tools.

The existing Makefile should work as advertised under MinGW/MSYS2. And the Windows-specific bug might no longer be an issue when using the latest library versions.

**Comment by RezaTaheri01** [comment]
I created a similar project by Python with the same functionality for Windows. Feel free to try it out :)

**Comment by Kamek437** [comment]
Awesome thanks man! Will definitely check it out.
